### PR TITLE
REP-4790 Exclude all Mongosync internal DBs, including those used by the embedded verifier

### PR DESCRIPTION
**Pull request by autarch**
This also includes some linting fixes and some tooling updates.

### CODE REVIEW

**Comment by autarch** [comment]
I ran the tests locally and they passed. This repo doesn't run PRs in CI :(

**Review by mmcclimon — CHANGES_REQUESTED** [review]
I think this is not quite right, and that we should a) add a test to confirm that it's not right (I suspect right now it's excluding the regex but not `admin` or something), and b) fix it. Thanks!

File: internal/verifier/list_namespaces.go [hunk 1]
```diff
@@ -29,9 +29,14 @@ func ListAllUserCollections(ctx context.Context, logger *logger.Logger, client *
...
 	excludedDBs = append(excludedDBs, additionalExcludedDBs...)
 	excludedDBs = append(excludedDBs, ExcludedSystemDBs...)
-	excludedDBs = append(excludedDBs, MongosyncMetaDB)
 
-	dbNames, err := client.ListDatabaseNames(ctx, bson.D{{"name", bson.D{{"$nin", excludedDBs}}}})
+	dbNames, err := client.ListDatabaseNames(
+		ctx,
+		bson.D{
+			{"name", bson.D{{"$nin", excludedDBs}}},
+			{"name", bson.D{{"$nin", []primitive.Regex{{Pattern: MongosyncMetaDBsPattern}}}}},
```
[the remark this answers is not in the document]
Comment (autarch) [reply internal/verifier/list_namespaces.go]: It _seemed_ to work, though I realized we weren't actually testing all the excludes. But even adding some more tests, it still seemed to work. But I fixed it anyway.

[the remark this answers is not in the document]
Comment (mmcclimon) [reply internal/verifier/list_namespaces.go]: Weird! I wonder if the driver does something weird here. Oh well, thanks all the same.

**Review by mmcclimon — APPROVED** [review]
LGTM, thanks!

**Pull request merged**